diff --git a/cmd-new-session.c b/cmd-new-session.c
index c90369bc..1abc2124 100644
--- a/cmd-new-session.c
+++ b/cmd-new-session.c
@@ -316,6 +316,10 @@ cmd_new_session_exec(struct cmd *self, struct cmdq_item *item)
 	}
 	notify_session("session-created", s);
 
+	if (!detached && !already_attached) {
+		notify_client("client-attached", c);
+	}
+
 	/*
 	 * Set the client to the new session. If a command client exists, it is
 	 * taking this session and needs to get MSG_READY and stay around.
diff --git a/input-keys.c b/input-keys.c
index 0451b968..c72d55b5 100644
--- a/input-keys.c
+++ b/input-keys.c
@@ -113,7 +113,7 @@ static struct input_key_entry input_key_defaults[] = {
 	  .data = "\033[5~"
 	},
 	{ .key = KEYC_BTAB,
-	  .data = "\033[Z"
+	  .data = "\033\011"
 	},
 
 	/* Arrow keys. */
