diff --git a/cmd-new-session.c b/cmd-new-session.c
index c90369bc..fcd57aac 100644
--- a/cmd-new-session.c
+++ b/cmd-new-session.c
@@ -342,8 +342,10 @@ cmd_new_session_exec(struct cmd *self, struct cmdq_item *item)
 		free(cp);
 	}
 
-	if (!detached)
+	if (!detached) {
 		c->flags |= CLIENT_ATTACHED;
+		notify_client("client-attached", c);
+	}
 	if (!args_has(args, 'd'))
 		cmd_find_from_session(current, s, 0);
 
diff --git a/input-keys.c b/input-keys.c
index 0451b968..c72d55b5 100644
--- a/input-keys.c
+++ b/input-keys.c
@@ -113,7 +113,7 @@ static struct input_key_entry input_key_defaults[] = {
 	  .data = "\033[5~"
 	},
 	{ .key = KEYC_BTAB,
-	  .data = "\033[Z"
+	  .data = "\033\011"
 	},
 
 	/* Arrow keys. */
